### ISSUE TITLE
Remapping `>` to `>gv` in visual mode breaks repeat (and deviates from vim)

### ISSUE THREAD

**Issue by Stebalien** [syl20bnr/spacemacs#5273]
#### Description
Spacemacs remaps`>` to `>gv` in visual mode  (`layers/+distribution/spacemacs-base/packages.el:286-287`). This

1. Breaks repeat (`v>.` doesn't indent twice).
2. Significantly deviates from vim and isn't documented anywhere.

#### Reproduction guide
- Start Emacs (evil mode)
- Type some text in the scratch buffer.
- From normal mode, enter `v>.` (enter visual mode, shift right, repeat).

*Observed behaviour:*
It claims that `.` is undefined. And only indents once.

*Expected behaviour:*
I expected it to indent twice.

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.11
- Spacemacs branch: master (rev. 93e55bb)
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(better-defaults git
                 (auto-completion :variables auto-completion-return-key-behavior nil)
                 org spell-checking syntax-checking
                 (shell :variables shell-default-shell 'ansi-term)
                 emacs-lisp markdown yaml c-c   html haskell go javascript python
                 (rust :variables rust-format-on-save t)
                 (latex :variables latex-build-command "LaTeX"))
```

**Comment by syl20bnr** [comment]
I believe there is a built-in function in Evil to do this.

**Comment by aldanor** [comment]
Looks like it's broken by Spacemacs indeed -- in isolated evil `v>.` indents things twice.

**Comment by StreakyCobra** [comment]
The problem is indeed on Spacemacs' side. Commenting out [these lines](https://github.com/syl20bnr/spacemacs/blob/f5924d1acb4c942fae1445b67014635609c60cae/layers/%2Bdistribution/spacemacs-base/packages.el#L294-L295) solves the problem.

They were added to:

>       ;; Keep the region active when shifting

From my point of view this should be removed to restore the repeating behavior, and if keeping the region active when shifting is needed it should be implemented upstream.

**Comment by Deradon** [comment]
> From my point of view this should be removed to restore the repeating behavior [...]

:+1:

**Comment by cpaulik** [comment]
Just for information, this behavior seems broken to me in Vim. 

starting with 

```
|test
test
```

Press `vj>..` in Vim I end up with 

```
      test
          test
```

whereas in spacemacs without the remapping mentioned here I end up with

```
      test
      test
```

**Comment by syl20bnr** [comment]
I made some research and this behavior is compatible with Vim.

1) `.` does nothing in visual state so the following code snippet to keep the selection active when indenting prevent to do `gv.`, this is the same behavior in Evil and Vim:

```
vnoremap > >gv
vnoremap < <gv
```

2) `v>.` is not necessary with the above snippet since you juste have to do `v>>`.

3) The snippet is more powerful: for instance `v><` is a no-op.

For this reasons I mark this issue as `won't fix`.

**Comment by Deradon** [comment]
That's strange.
Normal vim behaviour should result in the latter result.

`vj>......vj<......`:

![output](https://cloud.githubusercontent.com/assets/385500/14268776/ef49cab8-fade-11e5-86d8-7deca126f8e8.gif)

```
➜  ~ vim --version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Jan  2 2014 19:39:59)
```

**Comment by syl20bnr** [comment]
@Deradon simultaneous post :-)

**Comment by Deradon** [comment]
@syl20bnr Is there a way we can configure the behaviour? 

For me for example, it feels very unnatural to type `v>>>` compared to `v>..`
It just feels natural for me to use the dot command whenever I want to repeat something.

Pressing ">" requires pressing shift + key (at least for me), whereas pressing "." is way easier.

**Comment by syl20bnr** [comment]
We can define a dotfile variable `dotspacemacs-keep-selection-when-indenting` with an appropriate docstring.

**Comment by joehillen** [comment]
@syl20bnr I'll agree that this seems in the same category as `remap-Y-to-y$` (which seems to stopped working, btw. I'll open an issue.), even though I prefer the current behavior and I've had that binding set for almost the entire 6 years I've been using vim/evil.

I wonder if these kinds of variables should be moved outside the main `dotspacemacs-*` list into a separate layer. Maybe a `vim` layer?

**Comment by syl20bnr** [comment]
@joehillen there is a PR for a `common-tweaks` layer, seems to be a good place to put them indeed.

**Comment by cpaulik** [comment]
@Deradon I think I was using neovim with some config so maybe it is an issue with that. On another machine with vanilla vim it works like you showed.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
